back/app/controllers/web_api/v1/custom_pages_controller.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+class WebApi::V1::CustomPagesController < ApplicationController
+  skip_before_action :authenticate_user, only: %i[index show by_slug]
+  before_action :set_page, only: %i[show update destroy]
+
+  def index
+    @pages = paginate policy_scope(CustomPage)
+
+    render json: linked_json(@pages, WebApi::V1::CustomPageSerializer, params: fastjson_params)
+  end
+
+  def show
+    authorize @page
+    render json: WebApi::V1::CustomPageSerializer.new(@page, params: fastjson_params).serialized_json
+  end
+
+  def by_slug
+    @page = CustomPage.find_by! slug: params[:slug]
+    authorize @page
+    show
+  end
+
+  def create
+    @page = CustomPage.new permitted_attributes(CustomPage)
+    authorize @page
+
+    SideFxCustomPageService.new.before_create @page, current_user
+    if @page.save
+      SideFxCustomPageService.new.after_create @page, current_user
+      render(
+        json: WebApi::V1::CustomPageSerializer.new(@page, params: fastjson_params).serialized_json,
+        status: :created
+      )
+    else
+      render json: { errors: @page.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    assign_attributes_for_update
+    authorize @page
+
+    SideFxCustomPageService.new.before_update @page, current_user
+    if @page.save
+      SideFxCustomPageService.new.after_update @page, current_user
+      render json: WebApi::V1::CustomPageSerializer.new(@page, params: fastjson_params).serialized_json, status: :ok
+    else
+      render json: { errors: @page.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    page = @page.destroy
```

Comment:
The assign to a local variable looks a bit redundant here, couldn't we just continue to use the instance variable, as we do in the other actions as well?